agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/telemetry/MetricFilter.java

```diff
@@ -85,14 +72,6 @@ boolean matches(String metricName) {
   }
 
   public static boolean shouldSkip(String metricName, List<MetricFilter> metricFilters) {
-    if (!MetricFilter.NON_FILTERABLE_METRIC_NAMES.contains(metricName)) {
-      for (MetricFilter metricFilter : metricFilters) {
-        if (!metricFilter.matches(metricName)) {
-          // user configuration filtered out this metric name
-          return true;
-        }
-      }
-    }
-    return false;
+    return metricFilters.stream().anyMatch(metricFilter -> metricFilter.exclude(metricName));
```

Comment:
I have noticed that this method is called from a [loop](https://github.com/microsoft/ApplicationInsights-Java/blob/d195ed21b89124a1a98f0484ecc8928bf4debb9a/agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/exporter/AgentMetricExporter.java#L62) in the `AgentMetricExporter` class. If the number of iterations could be high, I would prefer to avoid creating a stream at each iteration, to limit heap allocation.